app/models/service/registry_api.rb

```diff
@@ -89,10 +89,37 @@ def tags_v2(image_name)
       get("/v2/#{image_name}/tags/list")['tags'].map { |tag| { 'name' => tag } }
     end
 
+    def connection_options
+      {}.merge(credentials)
+        .merge(http_proxy)
+    end
+
     def credentials
       { user: @user, password: @password }
     end
 
+    # Returns a proxy to set for connection_options
+    #
+    # If a proxy is set via ENV it will return and
+    # let Excon::Connection set it.
+    def http_proxy
+      if !env_http_proxy? && http_proxy_url
+        { proxy: http_proxy_url }
+      else
+        {}
+      end
+    end
+
+    def http_proxy_url
+      SETTINGS.fetch(:'foreman-docker', {})
```

Comment:
This is a bit problematic, as many users have multiple Foreman hosts - we're trying to move as much as we can out of settings.yaml. The reason is that it's less maintainable and requires a restart to change.

Could you move this to be a regular setting instead please?